src/components/Modal.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import { useMutation } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+
+const modalRoot = document.getElementById('modal');
+
+const ADD_VOLUNTEER = gql`
+  mutation addVolunteer($volunteerId: Int!, $eventId: Int!) {
+    addVolunteer(volunteerId: $volunteerId, eventId: $eventId) {
+      ok
+      errors {
+        path
+        message
+      }
+    }
+  }
+`;
+
+const Modal = ({ isModalOpen, modalContent, closeModal }: any) => {
+  const { title, location, date, nonprofit, id: eventId } = modalContent;
+
+  const [addVolunteer] = useMutation(ADD_VOLUNTEER);
+
+  const handleAttendClick = async () => {
+    const volunteerId = parseInt(localStorage.getItem('volunteerId')!);
+    let mutationResult = await addVolunteer({
+      variables: {
+        volunteerId,
+        eventId,
+      },
+    });
+    if (mutationResult.data.addVolunteer.ok) {
+      closeModal();
+      //add notfication of successful signup
+    }
+  };
+
+  const el = document.createElement('div');
+
+  useEffect(() => {
+    modalRoot?.appendChild(el);
```

Comment:
This should always be defined, and if it is not, things are broken.  Defensive programming hides real bugs.